### DISCUSSION TITLE
Error: "Axis X is outside the permitted range."

### DISCUSSION THREAD

**Discussion by pablodm00** [visose/Robots#120] [Q&A] [answered]
Hey, I have been using the plugin and keep getting this error. It happens when the values of the joint angles are high (usually above 2-3 rad), it says that these values are above the permitted range. However, these values are what are required and they should be doable by the robot as they are below 2*pi radians (a full rotation of the joint), right? 

am i not understanding something correctly? or is there a way to remove whatever limit is set (or change it to 2*pi?)

thanks in advance!

**Comment by visose** [comment, marked as answer]
Check the replies in this issue: https://github.com/visose/Robots/issues/8
I just pinned it, it seems a common misunderstanding people run into that is not immediately obvious.

**Comment by pablodm00** [comment]
Thanks for the quick help!

I had a look and its great to know but I'm not sure it answers my question. Also I was using that component to convert to radians already. Is there anything else that may be causing that?

**Comment by visose** [comment]
You mean the component to convert to Degrees? Are you using the one from the Robots plugin?

**Comment by pablodm00** [comment]
yes, i am using the robots plugin component to convert from degrees to radians

**Comment by visose** [comment]
Can you describe now in more detail the issue, for example:
- I am using robot model X
- The spec sheet says that it's 5th joint has a range from -180  to 180 degrees.
- Using the Robos plugin component to convert degrees to radians and after pluging the above values and the robot I'm using to the RobotSystem input, the range becomes: 0 to 2*PI radians.
- In grasshopper I get an error in axis 5 that the joint is at 1.5*PI radians.
- This value is within the 0 to 2*PI range, so the error is unexpected.

**Comment by pablodm00** [comment]
Thanks @visose, i checked and the value is indeed out of the allowed bounds. It happens in joint 5 of an ABB IRB1600 robot, which allows 115 to -115° but the value is -146°. why is this happening? Shouldn't the plugin not be generating angles outside of those permitted for that joint? the correct robot is connected to the program output, so I don't really understand why its doing this. Thanks in advance!

**Comment by visose** [comment]
It flags as an error that the angle is out of range. I think its better to display what the angle should be to reach that point, rather that not show anything at all, to help you decide how to fix it.

**Comment by pablodm00** [comment]
i see, thank you very much. What can i do to fix it? the object is at a normal place and there shouldn't be a problem reaching it

**Comment by visose** [comment]
There is a problem reaching it, at least with one configuration. You can try changing the configuration or see if an axis is getting winded up due to a previous motion.
